app/assets/javascripts/admin/variant_overrides/controllers/variant_overrides_controller.js.coffee

```diff
@@ -105,3 +109,31 @@ angular.module("admin.variantOverrides").controller "AdminVariantOverridesCtrl",
         StatusMessage.display 'success', t('js.variant_overrides.stock_reset')
       .error (data, status) ->
         $timeout -> StatusMessage.display 'failure', $scope.updateError(data, status)
+
+  $scope.countOnHandPlaceholder = (variant, hubId) ->
+    variantOverride = $scope.variantOverrides[hubId][variant.id]
+
+    if variantOverride.on_demand
+      t('js.variants.on_demand.yes')
+    else if variantOverride.on_demand == false
+      ''
+    else
+      variant.on_hand
```

Comment:
I see the tests are covering this but I'd still add a comment to this critical part of the code so that first time readers can follow.

I'd go for a method doc like this:
```js
// variant override count_on_hand field placeholder logic:
//     on_demand true  -- show "On Demand"
//     on_demand false -- show empty value to be set by the user
//     on_demand nil   -- show producer on_hand value
$scope.countOnHandPlaceholder = (variant, hubId) ->
    variantOverride = $scope.variantOverrides[hubId][variant.id]

      if variantOverride.on_demand == true
            t('js.variants.on_demand.yes')
      else if variantOverride.on_demand == false
            ''
      else
           variant.on_hand
```

Reply:
Done! :slightly_smiling_face: Thanks. https://github.com/openfoodfoundation/openfoodnetwork/pull/3104/commits/02c6f74bed966f80af9d6b9b06fe3e6d0cf4cf9d